### DISCUSSION TITLE
Yazi as Default File Viewer on mac os

### DISCUSSION THREAD

**Discussion by Sinled** [sxyazi/yazi#1828] [1. Q&A] [answered]
Is it possible? GUI file managers such as Forklift, [allows you](https://binarynights.com/manual#fileviewer) to replace Finder as system wide file manager on mac os, so it can be used in places where finder is usually opened. e.g. opening folders on desktop, dock, from spotlight etc.

**Comment by sxyazi** [comment, marked as answer]
I’m not really sure if this is possible, but Yazi is a terminal program, so it can't run independently outside of the terminal environment. 

So what you actually want to focus on might be: how to set your terminal to replace Finder and have Yazi run by default when the terminal starts

**Comment by Sinled** [comment]
So if anyone stumbles upon this, you can make it work. Here what i did with iTerm
1. Find terminal bundle intendifier `mdls -name kMDItemCFBundleIdentifier /Applications/YourApp.app` for iterm it is `com.googlecode.iterm2`
2. Set iterm as default file viewer
```
defaults write -g NSFileViewer -string com.googlecode.iterm2;
defaults write com.apple.LaunchServices/com.apple.launchservices.secure LSHandlers -array-add '{LSHandlerContentType="public.folder";LSHandlerRoleAll="com.googlecode.iterm2";}'
```
3. Restart system
4. Run `yazi` on iterm session start (I am using iTerm exclusively for yazi) `Profile > Command > send text at start`

you can return default file viewer with 
```
defaults delete -g NSFileViewer
defaults write com.apple.LaunchServices/com.apple.launchservices.secure LSHandlers -array-add '{LSHandlerContentType="public.folder";LSHandlerRoleAll="com.apple.finder";}'
```
